sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,201 @@
+      // and this mock must be removed.
+      const formService = await getProviderInstanceForModule(
+        appModule,
+        AppStudentsModule,
+        FormService,
+      );
+      const mockDryRunSubmission = jest.fn().mockResolvedValue({
+        valid: true,
+        formName: financialInformationFormName,
+        data: { data: financialInformationData },
+      });
+
+      formService.dryRunSubmission = mockDryRunSubmission;
+
+      const endpoint = `/students/appeal/application/${application.id}`;
+
+      // Act/Assert
+      let createdAppealId: number;
+      await request(app.getHttpServer())
+        .post(endpoint)
+        .send(payload)
+        .auth(studentToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.CREATED)
+        .then((response) => {
+          expect(response.body.id).toBeGreaterThan(0);
+          createdAppealId = +response.body.id;
+        });
+      const studentAppeal = await db.studentAppeal.findOne({
+        select: {
+          id: true,
+          appealRequests: {
+            id: true,
+            submittedFormName: true,
+            submittedData: true,
+          },
+        },
+        relations: { appealRequests: true },
+        where: { application: { id: application.id } },
+      });
+      const [appealRequest] = studentAppeal.appealRequests;
+      expect(studentAppeal.id).toBe(createdAppealId);
+      expect(appealRequest.submittedFormName).toBe(
+        financialInformationFormName,
+      );
+      expect(appealRequest.submittedData).toStrictEqual(
+        financialInformationData,
+      );
+      // Expect to call the dry run submission.
+      expect(mockDryRunSubmission).toHaveBeenCalledWith(
+        financialInformationFormName,
+        {
+          ...financialInformationData,
+          programYear: application.programYear.programYear,
+        },
+      );
+    },
+  );
+
+  it(
+    "Should throw Unprocessable Entity Exception when a student submit an appeal for financial information " +
+      "but already have a pending appeal to be completed.",
+    async () => {
+      // Arrange
+      // Create student to submit application.
+      const student = await saveFakeStudent(appDataSource);
+      // Create application to request change.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+        },
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+      // Create pending appeal for the student.
+      const pendingAppealRequest = createFakeStudentAppealRequest();
+      pendingAppealRequest.appealStatus = StudentAppealStatus.Pending;
+      const pendingAppeal = createFakeStudentAppeal({
+        application,
+        appealRequests: [pendingAppealRequest],
+      });
+      await db.studentAppeal.save(pendingAppeal);
+      // Prepare the data to request a change of financial information.
+      const financialInformationFormName = "studentfinancialinformationappeal";
```

Comment:
It can be defined as a const named `FINANCIAL_INFORMATION_FORM_NAME` at the file level.